Filtermenu buttons template (#1532)
telerik/blazor-docs: components/grid/templates/filter.md

![Custom Filter Menu Template with Checkboxes](images/custom-filter-menu-checkboxes.png)

## Filter Menu Buttons Template

> By default, the Filter Menu renders the `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 

Reviewer: ```suggestion
By default, the Filter Menu renders `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 
```

The template receives a `context` of type `FilterMenuTemplateContext` that provides the following members:

* `FilterDescriptor`—the object that describes the column filter. By default, the column filter has two filters: one for the type and another for the name of the field. You can modify the column filter by:
   * Adding more filters to the `FilterDescriptors` collection.
   * Changing the `LogicalOperator` (`AND` by default).
   * Using the `FilterDescriptor` to create a custom button that applies a predefined filter. 
* `FilterAsync`—applies the filters defined in the Filter Menu to the Grid component.
* `ClearFilterAsync`—clears the applied filters.

> >caption Implementing a button with a predefined filter.

Reviewer: Not sure what "predefined filter" is.

```suggestion
>caption Using custom filter menu buttons
```

![Custom Filter Menu Template with Checkboxes](images/custom-filter-menu-checkboxes.png)

## Filter Menu Buttons Template

By default, the Filter Menu renders the `Filter` and `Clear` buttons. You can customize or remove them entirely by using the `FilterMenuButtonsTemplate` tag. 

The template receives a `context` of type `FilterMenuTemplateContext` that provides the following members:

> * `FilterDescriptor` - the object that describes the column filter. By default it has two filters with the type and name of the field, and you can add more to its `FilterDescriptors` collection, or change its `LogicalOperator` from the default `AND`. You can use the filter descriptor to create a custom button that applies a predefined filter. 

Reviewer 2: Replace the pronouns with the actual nouns to improve the readability:
- what does "it" refer to ("By default it has two filters...")?
- "its `FilterDescriptors`",  "its `LogicalOperator`"

```suggestion
* `FilterDescriptor`—the object that describes the column filter. By default, the column filter has two filters: one for the type and another for the name of the field. You can modify the column filter by:
   * Adding more filters to the `FilterDescriptors` collection.
   * Changing the `LogicalOperator` (`AND` by default).
   * Using the `FilterDescriptor` to create a custom button that applies a predefined filter. 
```

> * `FilterAsync` - applies the defined filters in the Filter Menu to the Grid component.

Reviewer 2: ```suggestion
* `FilterAsync`—applies the filters defined in the Filter Menu to the Grid component.
```

> * `ClearFilterAsync` - clears the applied filters.

Reviewer 2: ```suggestion
* `ClearFilterAsync`—clears the applied filters.
```

> >caption Implement a button with predefined filter.

Reviewer 2: ```suggestion
>caption Implementing a button with a predefined filter.
```
